src/runtime_src/core/tools/xbmgmt2/SubCmdFlash.cpp

```diff
@@ -46,8 +47,83 @@ namespace {
 static unsigned int
 bdf2index()
 {
-  //this should be placed in xbmgmt common
-  return 0;
+  return 0;//hardcoded for now
+}
+static void scan_devices(bool verbose, bool json)
+{
+  verbose = verbose;
+  json = json;
+
+  Flasher f(bdf2index());
+  if (!f.isValid())
+      return;
+
+  DSAInfo board = f.getOnBoardDSA();
+  std::vector<DSAInfo> installedDSA = f.getInstalledDSA();
+  BoardInfo info;
+  f.getBoardInfo(info);
+  std::cout << "Card [" << bdf2index() << "]\n";
+  std::cout << "\tCard type:\t\t" << board.board << "\n";
+  std::cout << "\tFlash type:\t\t" << f.sGetFlashType() << "\n";
+  std::cout << "\tFlashable partition running on FPGA:" << "\n";
+  std::cout << "\t\t" << board << "\n";
+  std::cout << "\tCard name\t\t\t" << info.mName << "\n";
+  std::cout << "\tCard S/N: \t\t\t" << info.mSerialNum << "\n";
+  std::cout << "\tConfig mode: \t\t" << info.mConfigMode << "\n";
+  std::cout << "\tFan presence:\t\t" << info.mFanPresence << "\n";
+  std::cout << "\tMax power level:\t\t" << info.mMaxPower << "\n";
+  std::cout << "\tMAC address0:\t\t" << info.mMacAddr0 << "\n";
+  std::cout << "\tMAC address1:\t\t" << info.mMacAddr1 << "\n";
+  std::cout << "\tMAC address2:\t\t" << info.mMacAddr2 << "\n";
+  std::cout << "\tMAC address3:\t\t" << info.mMacAddr3 << "\n";
+
+}
+
+// Update shell on the board.
+static void update_shell(unsigned index, std::string flashType,

```

Comment:
unsigned int, const std::string& flashType
Personally I don't like "unsigned" alone. unsigned what?

Reply:
Thanks for pointing it out. I fixed this is most of the places, might have missed a couple.